### PR TITLE
Add DiagnosticsPlots class with 6 time-series diagnostic methods

### PR DESCRIPTION
**Pull request by isofer**
## Description

This PR adds the `DiagnosticsPlots` class to `pymc_marketing.mmm.plotting`, built on top of the foundation helpers from `isofer/pr1-foundation-helpers` (arviz-plots integration, `_extract_matplotlib_result`, `_process_plot_params`, `_validate_dims`, `channel_color_map`).

`DiagnosticsPlots` exposes six methods for inspecting MMM model fit:

1. `posterior_predictive` — Mean + HDI time series vs. observed target.
2. `prior_predictive` — Same layout using prior predictive samples.
3. `residuals_over_time` — Residuals (target − predictions) over time.
4. `residuals_distribution` — Posterior distribution of residuals via arviz-plots.
5. `posterior` — 1-D marginal KDE of posterior variables.
6. `prior_vs_posterior` — Overlaid prior and posterior KDE distributions.

See [this demo notebook](https://gist.github.com/isofer/7d1fcaf6db67c450cefdb5e985bf21af) for examples.

## Related Issue
Related to #2458 

## Changes Made (vs `isofer/pr1-foundation-helpers`)

- **`pymc_marketing/mmm/plotting/diagnostics.py`** (new, 895 lines): `DiagnosticsPlots` class + module-level idata accessor helpers (`_get_posterior_predictive`, `_get_prior_predictive`, `_get_prior`, `_get_prior_for_plot`).
- **`pymc_marketing/mmm/plotting/__init__.py`**: Exports `DiagnosticsPlots`.
- **`pymc_marketing/mmm/plotting/_helpers.py`**: Refactored `_extract_matplotlib_result` to use `fig.get_axes()`; added note about the `col_wrap` edge case.
- **`tests/mmm/plotting/test_diagnostics.py`** (new, 1092 lines): Full test suite covering both the simple (no extra dims) and multi-dim fixtures, all six plot methods, `original_scale` flag, error paths, and `return_as_pc`.
- **`tests/mmm/plotting/test_helpers.py`**: Extended with tests for the updated `_extract_matplotlib_result`.

## Screenshots / Examples

```python
from pymc_marketing.mmm.plotting import DiagnosticsPlots

dp = DiagnosticsPlots(mmm.idata_wrapper)

# Posterior predictive check (transformed scale)
fig, axes = dp.posterior_predictive(original_scale=False)

# Prior vs posterior for named variables
fig, axes = dp.prior_vs_posterior(var_names=["alpha", "sigma"])

# Residuals over time
fig, axes = dp.residuals()
```

----
📚 Documentation preview 📚: https://pymc-marketing--2459.org.readthedocs.build/en/2459/

### CODE REVIEW

File: pymc_marketing/mmm/plotting/_helpers.py [hunk 1]
```diff
@@ -196,9 +196,6 @@ def _extract_matplotlib_result(
     """
     if return_as_pc:
         return pc
-
     fig = pc.viz.ds["figure"].item()
-    plot_da = pc.viz.ds["plot"]
-    axes = np.atleast_1d(np.array(plot_da.values.flat))
-
+    axes = np.atleast_1d(np.array(fig.get_axes()))
```
[the remark this answers is not in the document]
Comment (isofer) [reply pymc_marketing/mmm/plotting/_helpers.py:200]: wrong

**Comment by isofer** [comment]
@juanitorduz this is also ready for review, and can be merged after #2437

**Comment by isofer** [comment]
@juanitorduz ready for review

File: pymc_marketing/mmm/plotting/diagnostics.py [hunk 2]
```diff
@@ -0,0 +1,970 @@
...
+from pymc_marketing.data.idata import MMMIDataWrapper
+from pymc_marketing.mmm.plotting._helpers import (
+    _dims_to_sel_kwargs,
+    _extract_matplotlib_result,
+    _process_plot_params,
+    _select_dims,
+)
+
+
+def _get_posterior_predictive(data: MMMIDataWrapper) -> xr.Dataset:
```
Comment (juanitorduz) [inline pymc_marketing/mmm/plotting/diagnostics.py]: Can we put this here https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/data/idata/utils.py ? (maybe not a private function, so we remove the "_" prefix ?

File: pymc_marketing/mmm/plotting/diagnostics.py [hunk 3]
```diff
@@ -0,0 +1,970 @@
...
+        or data.idata.posterior_predictive is None
+    ):
+        raise ValueError(
+            "No posterior_predictive data found in idata. "
+            "Run MMM.sample_posterior_predictive() first."
+        )
+    return data.idata.posterior_predictive
+
+
+def _get_prior_predictive(data: MMMIDataWrapper) -> xr.Dataset:
```
Comment (juanitorduz) [inline pymc_marketing/mmm/plotting/diagnostics.py]: Same, put in https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/data/idata/utils.py

File: pymc_marketing/mmm/plotting/diagnostics.py [hunk 4]
```diff
@@ -0,0 +1,970 @@
...
+        or data.idata.prior_predictive is None
+    ):
+        raise ValueError(
+            "No prior_predictive data found in idata. "
+            "Run MMM.sample_prior_predictive() first."
+        )
+    return data.idata.prior_predictive
+
+
+def _get_prior(data: MMMIDataWrapper) -> xr.Dataset:
```
Comment (juanitorduz) [inline pymc_marketing/mmm/plotting/diagnostics.py]: same, put https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/data/idata/utils.py

File: tests/mmm/plotting/test_diagnostics.py [hunk 5]
```diff
@@ -0,0 +1,1092 @@
...
+@pytest.fixture(scope="module")
+def dist_plots(posterior_idata) -> DiagnosticsPlots:
+    """DiagnosticsPlots backed by posterior_idata (no MMMIDataWrapper validation)."""
+    wrapper = MMMIDataWrapper(posterior_idata, validate_on_init=False)
+    return DiagnosticsPlots(wrapper)
+
+
+# ============================================================================
+# Helper tests
+# ============================================================================
```
Comment (juanitorduz) [inline tests/mmm/plotting/test_diagnostics.py]: remove these comments (from these test files)

File: tests/mmm/plotting/test_diagnostics.py [hunk 6]
```diff
@@ -0,0 +1,1092 @@
...
+
+class TestDiagnosticsPlotsConstructor:
+    def test_stores_data(self, simple_data):
+        plots = DiagnosticsPlots(simple_data)
+        assert plots._data is simple_data
+
+
+# ============================================================================
+# posterior_predictive tests
+# ============================================================================
```
Comment (juanitorduz) [inline tests/mmm/plotting/test_diagnostics.py]: same (and all the ones below)

**Review by juanitorduz — CHANGES_REQUESTED** [review]
This looks great! Thanks! some minor comments and we are good to go!

File: pymc_marketing/mmm/plotting/diagnostics.py (hunk 2 above)
Comment (isofer) [reply pymc_marketing/mmm/plotting/diagnostics.py]: Done.

File: pymc_marketing/mmm/plotting/diagnostics.py (hunk 3 above)
Comment (isofer) [reply pymc_marketing/mmm/plotting/diagnostics.py]: Done.

File: pymc_marketing/mmm/plotting/diagnostics.py (hunk 4 above)
Comment (isofer) [reply pymc_marketing/mmm/plotting/diagnostics.py]: Done.

File: tests/mmm/plotting/test_diagnostics.py (hunk 5 above)
Comment (isofer) [reply tests/mmm/plotting/test_diagnostics.py]: Done.

File: tests/mmm/plotting/test_diagnostics.py (hunk 6 above)
Comment (isofer) [reply tests/mmm/plotting/test_diagnostics.py]: Done.

**Comment by isofer** [comment]
@juanitorduz I've addressed all comments

**Review by juanitorduz — APPROVED** [review]
amazing! thanks!
